src/env.c

```diff
@@ -72,7 +68,7 @@ static struct {
         size_t image_size;
         // The size needed to store the full TLS.
         size_t memory_size;
-        // The size needed to store the TLS and the thread structure.
+        // The size needed to store the TLS and any library-defined thread metadata.
```

Comment:
This is not correct. If there is library-defined metadata this is not included in this size. This size only includes the size needed for the TLS region and the adjacent pair of pointers.

Reply:
I'm confused as to what those pointers are or what they're for.  I though it was for MUSLC's thread metadata.  I don't know what the 'thread structure' is---it's never defined here as far as I can see.

Reply:
The System-V ELF specification requires those pointers for the extended forms of TLS. sel4runtime only implements static TLS but if you want to include thread-local registers from dynamically loaded libraries, those pointers are used to navigate portions of the TLS region that are added dynamically